prov/tcp/src/tcpx.h

```diff
@@ -158,4 +164,12 @@ struct tcpx_fabric {
 	pthread_t		conn_mgr_thread;
 };
 
+struct tcpx_domain {
+	struct util_domain	util_domain;
+	struct tcpx_progress	*progress;
```

Comment:
Embed struct tcpx_progress directly, rather than storing a pointer.  The progress init function should expect the structure to already be allocated.  This will avoid additional memory allocations.